pages/cart.js

```diff
@@ -0,0 +1,251 @@
+import { useContext, useEffect, useState } from "react";
+import styled from "styled-components";
+import { toast } from "react-toastify";
+import { FaTimes } from "react-icons/fa";
+
+import Header from "../components/Header";
+import Footer from "../components/Footer";
+import Divider from "../styles/UIKit/Divider";
+import UIBUtton from "../styles/UIKit/Button";
+import useApi from "../utils/useApi";
+import CartConstants from "../config/CartConfig";
+import AuthContext from "../contexts/AuthContext";
+
+const Cart = () => {
+  const api = useApi();
+  const [cartItems, setCartItems] = useState([]);
+  const [cartData, setCartData] = useState({});
+  const [cartTotal, setCartTotal] = useState(0);
+  const { cart, updateCart } = useContext(AuthContext);
+
+  const fetchData = async () => {
+    try {
+      console.log("cart data", cart);
+
+      setCartData(cart);
+      setCartTotal(cart.total_price);
+      setCartItems(cart.items);
+    } catch (err) {
+      toast.error("Cannot fetch cart data");
+      console.log("Error while fetching cart data : ", err);
+    }
+  };
+
+  const handleDeleteCartItem = (itemId) => {
+    deleteCartItem(itemId);
+  };
+
+  const deleteCartItem = async (itemId) => {
+    const newCartItems = cartData.items.filter(
```

Comment:
No need to do all this, the api returns the updated cart, you can direct update cart with the response from the api

Reply:
does the api return the items.product populate content also?

Reply:
taken care

Reply:
taken care